pkg/tsdb/loki/loki.go

```diff
@@ -166,6 +196,49 @@ func (s *Service) QueryData(ctx context.Context, req *backend.QueryDataRequest)
 	return queryData(ctx, req, dsInfo, responseOpts, s.tracer, logger, isFeatureEnabled(ctx, featuremgmt.FlagLokiRunQueriesInParallel), isFeatureEnabled(ctx, featuremgmt.FlagLokiStructuredMetadata))
 }
 
+func getDashboardPanelNames(ctx context.Context, dshUid string, panelId string, s *Service, logger log.Logger) (*dashboardPanelNames, error) {
+	if dshUid == "" || panelId == "" {
+		return nil, fmt.Errorf("dashboard uid or panel id is empty")
+	}
+
+	dashboardData, err := s.dashService.GetDashboard(ctx, &dashboards.GetDashboardQuery{UID: dshUid})
```

Comment:
🤔 hymm -- this will make running loki as a multi-tenant service pretty complicated

I don't like it -- but shoving this in from the frontend seems better